resource-managers/kubernetes/core/src/test/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodFactorySuite.scala

```diff
@@ -0,0 +1,349 @@
+      "1", "dummy", "dummy", Seq[(String, String)](), driverPod, Map[String, Int]())
+
+    verify(nodeAffinityExecutorPodModifier, times(1))
+      .addNodeAffinityAnnotationIfUseful(any(classOf[Pod]), any(classOf[Map[String, Int]]))
+
+    assert(executor.getMetadata.getAnnotations.size() === 1)
+    assert(executor.getMetadata.getAnnotations.containsKey(constants.INIT_CONTAINER_ANNOTATION))
+    checkOwnerReferences(executor, driverPodUid)
+  }
+
+  test("the shuffle-service adds a volume mount") {
+    val conf = baseConf.clone()
+    conf.set(KUBERNETES_SHUFFLE_LABELS, "label=value")
+    conf.set(KUBERNETES_SHUFFLE_NAMESPACE, "default")
+    conf.set(KUBERNETES_SHUFFLE_DIR, "/tmp")
+
+    val shuffleManager = mock(classOf[KubernetesExternalShuffleManager])
+    when(shuffleManager.getExecutorShuffleDirVolumesWithMounts).thenReturn({
+      val shuffleDirs = Seq("/tmp")
+      shuffleDirs.zipWithIndex.map { case (shuffleDir, shuffleDirIndex) =>
+        val volumeName = s"$shuffleDirIndex-${FilenameUtils.getBaseName(shuffleDir)}"
+        val volume = new VolumeBuilder()
+          .withName(volumeName)
+          .withNewHostPath(shuffleDir)
+          .build()
+        val volumeMount = new VolumeMountBuilder()
+          .withName(volumeName)
+          .withMountPath(shuffleDir)
+          .build()
+        (volume, volumeMount)
+      }
+    })
+
+    val factory = new ExecutorPodFactoryImpl(
+      conf,
+      nodeAffinityExecutorPodModifier,
+      None,
+      None,
+      None,
+      None,
+      Some(shuffleManager))
+    val executor = factory.createExecutorPod(
+      "1", "dummy", "dummy", Seq[(String, String)](), driverPod, Map[String, Int]())
+
+    verify(nodeAffinityExecutorPodModifier, times(1))
+      .addNodeAffinityAnnotationIfUseful(any(classOf[Pod]), any(classOf[Map[String, Int]]))
+
+    assert(executor.getSpec.getContainers.size() === 1)
+    assert(executor.getSpec.getContainers.get(0).getVolumeMounts.size() === 1)
+    assert(executor.getSpec.getContainers.get(0).getVolumeMounts.get(0).getName === "0-tmp")
+    assert(executor.getSpec.getContainers.get(0).getVolumeMounts.get(0)
+      .getMountPath === "/tmp")
+    checkOwnerReferences(executor, driverPodUid)
+  }
+
+  test("Small-files add a secret & secret volume mount to the container") {
+    val conf = baseConf.clone()
+
+    //val smallFiles = new MountSmallFilesBootstrapImpl("secret1", "/var/secret1")
+    val smallFiles = mock(classOf[MountSmallFilesBootstrap])
+    when(smallFiles.mountSmallFilesSecret(
+      any(classOf[Pod]),
+      any(classOf[Container]))).thenAnswer(new Answer[(Pod, Container)] {
+        def answer(invocation: InvocationOnMock): (Pod, Container) = {
+          val pod = invocation.getArgumentAt(0, classOf[Pod])
+          val container = invocation.getArgumentAt(1, classOf[Container])
+          val secretName = "secret1"
+          val secretMountPath = "/var/secret1"
+          val resolvedPod = new PodBuilder(pod)
```

Comment:
These can be simplified. Just return the pod with an extra label/annotation, and the container with an extra environment variable. Verify that the pod has this extra label/annotation and the container has the appropriate environment variable.

Or, use `Mockito.verify()` - I'm still not sure which is the stronger/more idiomatic approach here.

Reply:
I don't like that we're mocking the very behavior we're testing here. I'd actually be in favor of omitting mocks for this particular case entirely and just relying on a real stamped out `MountSmallFilesBootstrap` since it's not nearly complex enough to need a mock IMO.